File: test_data/code/java/1287-element-appearing-more-than-25-in-sorted-array.java
class Solution {
    public int findSpecialInteger(int[] arr) {
        int n = arr.length;
        int targetCount = (int)Math.ceil(n * 0.25);
        for (int i = 0; i < n; i++) {
            int count = 1;
            for (int j = i + 1; j < n; j++) {
                if (arr[i] == arr[j]) {
                    count++;
                } else {
                    break;
                }
            }
            if (count >= targetCount) {
                return arr[i];
            }
        }
        return -1;
    }
}

File: test_data/code/python/1287-element-appearing-more-than-25-in-sorted-array.py
class Solution:
    def findSpecialInteger(self, arr: List[int]) -> int:
        n = len(arr)
        target_count = n // 4
        for i, num in enumerate(arr):
            count = sum(1 for j in range(i, n) if arr[j] == num)
            if count > target_count:
                return num